sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java

```diff
@@ -197,4 +203,21 @@ private ArrayList<TokenCredential> getCredentialsChain() {
         output.add(new AzurePowerShellCredential(tenantId, identityClientOptions.clone()));
         return output;
     }
+
+    private WorkloadIdentityCredential getWorkloadIdentityCredentialIfAvailable() {
+        Configuration configuration = identityClientOptions.getConfiguration() == null
+            ? Configuration.getGlobalConfiguration().clone() : identityClientOptions.getConfiguration();
+
+        String tenantId = configuration.get(Configuration.PROPERTY_AZURE_TENANT_ID);
+        String federatedTokenFilePath = configuration.get(AZURE_FEDERATED_TOKEN_FILE);
+        String azureAuthorityHost = configuration.get(Configuration.PROPERTY_AZURE_AUTHORITY_HOST);
+        if (!(CoreUtils.isNullOrEmpty(tenantId)
+            || CoreUtils.isNullOrEmpty(federatedTokenFilePath)
+            || CoreUtils.isNullOrEmpty(managedIdentityClientId)
```

Comment:
Should we check whether `managedIdentityClientId` is null or empty before doing any of the checks as it's the only constructor parameter not retrieved in this method.

Reply:
yeah, this can be done, added it.